### PR TITLE
Design architecture

### LINKED ISSUE #4: Design architecture
Sketch out repo structure, packages structure, class and method names.

We should aim for test driven development with docstrings for all methods & general doc suitable for publicising on PyPi.

### PR DESCRIPTION
**Pull request by ubyndr**
Resolves #4 

I have added classes and methods, and empty test classes.

I'm not 100% sure about the curie validation part, class hierarchy and static methods I've added. It would be nice to have a discussion.

### CODE REVIEW

File: src/pandasaurus/query.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        # Not sure about this
+        self.enriched_df = pd.DataFrame()
+
+    def simple_slim_enrichment(self):
+        logging.info(self.seed_list)
+        df = pd.DataFrame()
+        # TODO Add missing implementation
+        self.enriched_df = df
+
+    def minimal_slim_enrichment(self):
```
Comment (dosumis) [inline src/pandasaurus/query.py:28]: This and full slim enrichment need an arg for users to specify a list of slims.

File: src/pandasaurus/query.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    """
+    Query class
+    """
+
+    def __init__(self, seed_list: List[str]):
+        self.seed_list = seed_list
+        # Not sure about this
+        self.enriched_df = pd.DataFrame()
+
+    def simple_slim_enrichment(self):
```
Comment (dosumis) [inline src/pandasaurus/query.py:22]: doesn't involve slims.  Name this 'simple_enrichment'

**Review by dosumis — CHANGES_REQUESTED** [review]
Looks great.  Some small changes requested.  Would like to get one  or two more reviews @anitacaron  @udp @hkir-dev would you have time to take a look? (this is to fulfill functional spec in #1 )

File: src/pandasaurus/query.py [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+        # TODO Add missing implementation
+        self.enriched_df = df
+
+    def full_slim_enrichment(self):
+        logging.info(self.seed_list)
+        df = pd.DataFrame()
+        # TODO Add missing implementation
+        self.enriched_df = df
+
+    def contextual_slim_enrichment(self):
```
Comment (dosumis) [inline src/pandasaurus/query.py:40]: Needs context as arg.  Context should be a list of triples to use in SPARQL query of ubergraph.

**Comment by dosumis** [comment]
> Are we going to use poetry?

@anitacaron - Seems reasonable, but maybe not needed on this PR - can you create a ticket & link to the details?

File: src/pandasaurus/utils/query_utils.py [hunk 4]
```diff
@@ -0,0 +1,71 @@
+from ..config import default_config
+from typing import List
+from SPARQLWrapper import SPARQLWrapper, JSON
```
Comment (dosumis) [inline src/pandasaurus/utils/query_utils.py:3]: Consider using OAK as a wrapper for querying UberGraph.  There are some advantages  - e.g. support for standard prefixes.  @anitacaron  can tell you more.

**Comment by anitacaron** [comment]
> Add requests.txt with dependencies at root

We're going to use poetry for this. #§

File: src/pandasaurus/utils/query_utils.py (hunk 4 above)
Comment (ubyndr) [reply src/pandasaurus/utils/query_utils.py:3]: Even thought I'm not sure how the prefix validation will be done, do we really need an excavator for a task that can be done with a small shovel?

**Comment by hkir-dev** [comment]
In general project structure looks great. 

Can we make automatic code coverage tests part of CI? Recently I tried [Codecov](https://about.codecov.io/). It's free for public repos. You can configure it to run as part of your CI github action ([example action](https://github.com/hkir-dev/brain_data_standards_ontologies/blob/master/.github/workflows/build.yaml)). It will automatically calculate the code coverage change for each pull request and make a comment ([such as](https://github.com/hkir-dev/brain_data_standards_ontologies/pull/1#issuecomment-1522109951)).

Also it would be wonderful if we can find similar apps for code style and quality checks.

File: src/pandasaurus/query.py [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+    """
+    Query class
+    """
+
+    def __init__(self, seed_list: List[str]):
+        self.seed_list = seed_list
+        # Not sure about this
+        self.enriched_df = pd.DataFrame()
+
+    def simple_enrichment(self):
```
Comment (dosumis) [inline src/pandasaurus/query.py:22]: Add docstrings to methods based on functional spec.

**Comment by ubyndr** [comment]
> Can we make automatic code coverage tests part of CI?

I have created an issue for this, #8 .

**Comment by anitacaron** [comment]
> Also it would be wonderful if we can find similar apps for code style and quality checks.

I'll set up poetry with flask8 and add GH Action to check each PR.

**Comment by anitacaron** [comment]
@ubyndr, qc didn't pass 😬 let me know if you don't get the suggestion message.

**Comment by ubyndr** [comment]
> @ubyndr, qc didn't pass 😬 let me know if you don't get the suggestion message.

Thanks for the commits, but I would prefer the qc and poetry additions in an other PR. It would be better, wouldn't it?

**Comment by anitacaron** [comment]
The sooner you can start using QC, the better. As you can see, the QC is already not passing.

**Comment by ubyndr** [comment]
> The sooner you can start using QC, the better. As you can see, the QC is already not passing.

Fair point, but it is not in the scope of #6 . I made some updates but ignoring `F401` errors for now.

**Pull request merged**